Source/Factories/ServiceProviderFactory.cs

```diff
@@ -0,0 +1,24 @@
+using Microsoft.Extensions.DependencyInjection;
+using Common;
+
+namespace Source.Factories
+{
+    public static class ServiceProviderFactory
+    {
+        private static ServiceProvider CreateServiceProvider()
+        {
+            // configure and register services in container
+            return new ServiceCollection()
+                .AddMyAppServices()
+                .BuildServiceProvider();
+        }
+
+        public static IEnumerable<T> GetRequiredService<T>()
```

Comment:
Planning to add different types of interfaces?

Reply:
Yes, I am. One for LeetCode and one for Hackerrank.